dev_assistant_client/device.py

```diff
@@ -40,53 +47,44 @@ def create_device_payload():
         indent=4,
     )
 
-
-CONN = http.client.HTTPSConnection(APP_URL, cert_file=CERT_FILE, key_file=KEY_FILE)
-HEADERS = {
-    "Accept": "application/json",
-    "Content-Type": "application/json",
-}
-
-
-async def connect():
-    with open(TOKEN_FILE, "r") as f:
-        token = f.readline()
-
-    HEADERS["Authorization"] = "Bearer " + token
-
+api_client = APIClient(f"{APP_URL}/{API_PATH}", CERT_FILE, KEY_FILE)
+
+async def connect_device():
+    """
+    Tries to connect the device to the server. It starts by reading a token, creates
+    a device payload and makes a POST call to the server. If the call is successful,
+    the device is connected and the device ID is saved locally. Then, it tries to establish
+    a WebSocket connection using the ably_connect() function from auth.py.
+    """
+    auth = Auth()
+    if not auth.login():
+        print("Login failed. Exiting...")
+        return
+    
+    token = read_token()
+    
     payload = create_device_payload()
-
     print(now(), "Connecting...", sep="\t")
 
-    CONN.request("POST", API_PATH + "/devices", body=payload, headers=HEADERS)
-    response = CONN.getresponse()
-
-    if response.status == 200 or response.status == 201:
-        response_body = response.read().decode()
-        device_data = json.loads(response_body)
-        print(
-            now(),
-            "Connected.",
-            "Device ID " + Fore.LIGHTYELLOW_EX + device_data["id"] + Style.RESET_ALL,
-            sep="\t",
-        )
+    api_client.headers["Authorization"] = "Bearer " + token
+    
+    response = api_client.post("/devices", data=payload)
+    
+    if response.status_code in [200, 201]:
+        print(now(),"Connected.","Device ID " + Fore.LIGHTYELLOW_EX + json.loads(response.content).get("id") + Style.RESET_ALL,sep="\t")
         with open(DEVICE_ID_FILE, "w") as f:
-            f.write(device_data["id"])
-        # Connect to Ably
-        await ably_connect()
+            f.write(json.loads(response.content).get("id"))
+        await Auth().ably_connect()
     else:
         print(now(), "Failed to connect!", sep="\t")
-        if response.status == 401:
-            print(now(), "Error: ", response.read().decode(), sep="\t")
+        if response.status_code == 401:
+            print(now(), "Error: ", json.loads(response.content).get('error'), sep="\t")
```

Comment:
This probably is 'message' with a 'errors' object. Can you please check @lucianotonet ?